db/crud.go

```diff
@@ -2249,18 +2252,19 @@ func (db *DatabaseCollectionWithUser) updateAndReturnDoc(ctx context.Context, do
 
 			// Return the new raw document value for the bucket to store.
 			doc.SetCrc32cUserXattrHash()
-			var rawXattr, rawDocBody []byte
-			rawDocBody, rawXattr, err = doc.MarshalWithXattr()
+
+			var rawSyncXattr, rawVvXattr, rawDocBody []byte
+			rawDocBody, rawSyncXattr, rawVvXattr, err = doc.MarshalWithXattrs()
```

Comment:
Take or leave this because. the rest of the code seems fine and this is existing behavior. Conceptually, I'd expect an immediate return here.